addons/point_of_sale/static/src/app/store/pos_store.js

```diff
@@ -952,68 +956,84 @@ export class PosStore extends Reactive {
     cashierHasPriceControlRights() {
         return !this.config.restrict_price_control || this.get_cashier()._role == "manager";
     }
-    generate_unique_id() {
-        // Generates a public identification number for the order.
-        // The generated number must be unique and sequential. They are made 12 digit long
-        // to fit into EAN-13 barcodes, should it be needed
-
-        function zero_pad(num, size) {
-            var s = "" + num;
-            while (s.length < size) {
-                s = "0" + s;
-            }
-            return s;
-        }
-        return (
-            zero_pad(this.session.id, 5) +
-            "-" +
-            zero_pad(this.session.login_number, 3) +
-            "-" +
-            zero_pad(this.session.sequence_number, 4)
-        );
-    }
     createNewOrder(data = {}) {
+        if (!(data.pos_reference && data.sequence_number && data.tracking_number)) {
+            throw new Error("pos_reference, sequence_number and tracking_number are required");
+        }
+
         const fiscalPosition = this.models["account.fiscal.position"].find((fp) => {
             return fp.id === this.config.default_fiscal_position_id?.id;
         });
 
-        const uniqId = this.generate_unique_id();
         const order = this.models["pos.order"].create({
             session_id: this.session,
             company_id: this.company,
             config_id: this.config,
             picking_type_id: this.pickingType,
             user_id: this.user,
-            sequence_number: this.session.sequence_number,
             access_token: uuidv4(),
             ticket_code: random5Chars(),
             fiscal_position_id: fiscalPosition,
-            name: _t("Order %s", uniqId),
-            pos_reference: uniqId,
             ...data,
         });
 
-        this.session.sequence_number++;
         order.set_pricelist(this.config.pricelist_id);
+        order.recomputeOrderData();
+
         return order;
     }
-    add_new_order(data = {}) {
+    async add_new_order(data = {}) {
         if (this.get_order()) {
             this.get_order().updateSavedQuantity();
         }
-
+        Object.assign(data, await this.getNextOrdereRefs());
         const order = this.createNewOrder(data);
         this.selectedOrderUuid = order.uuid;
         this.searchProductWord = "";
         return order;
     }
-
+    async getNextOrdereRefs() {
```

Comment:
```suggestion
    async getNextOrdereRefs(newlyCreatedOrder) {
```

By passing the order object here, others methods can stay synchrone.

Just do something like:

```
const pos_ref = await rpc(...);
order.pos_ref = pos_ref;
```